### PR TITLE
[BleBox] fixes for gatebox and switchboxd

### PR DESCRIPTION
**Pull request by szczukot**

### CODE REVIEW

File: hardware/BleBox.cpp [hunk 1]
```diff
@@ -17,11 +17,9 @@ const _STR_DEVICE DevicesType[TOT_TYPE] =
 	{ 1, "shutterBox", "Shutter Box", int(pTypeLighting2), int(sTypeAC), int(STYPE_BlindsPercentageInverted), "shutter" },
 	{ 2, "wLightBoxS", "Light Box S", int(pTypeLighting2), int(sTypeAC), int(STYPE_Dimmer), "light" },
 	{ 3, "wLightBox", "Light Box", int(pTypeLimitlessLights), int(sTypeLimitlessRGBW), int(STYPE_Dimmer), "rgbw" },
-	{ 4, "gateBox", "Gate Box", int(pTypeLighting2), int(sTypeAC), int(STYPE_Dimmer), "gate" },
+	{ 4, "gateBox", "Gate Box", int(pTypeGeneral), int(sTypePercentage), 0, "gate" },
```
Comment (gizmocuz) [inline hardware/BleBox.cpp]: could you remove all 'int' casts from this block ? It's not needed and it makes it quite unreadable

File: hardware/BleBox.cpp [hunk 2]
```diff
@@ -273,192 +270,249 @@ bool BleBox::WriteToHardware(const char *pdata, const unsigned char length)
...
+					else
+					{
+						state = "0";
+					}
 
-				Json::Value root = SendCommand(IPAddress, "/s/" + state);
-				if (root == "")
-					return false;
+					Json::Value root = SendCommand(IPAddress, "/s/" + state);
+					if (root == "")
```
Comment (gizmocuz) [inline hardware/BleBox.cpp]: Did you intend to do if (root.empty()) ?

File: hardware/BleBox.cpp (hunk 1 above)
Comment (szczukot) [reply hardware/BleBox.cpp]: You're right

File: hardware/BleBox.cpp (hunk 2 above)
Comment (szczukot) [reply hardware/BleBox.cpp]: sure-it is better way

**Comment by gizmocuz** [comment]
Thanks!

**Pull request merged**